frontend/src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/components/Menu/index.tsx

```diff
@@ -1,86 +1,76 @@
+import { Props } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/components/Menu/types";
+import DeleteDataset from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/DeleteDataset";
 import {
-  MenuItemProps,
-  Intent,
-  Menu as RawMenu,
-  MenuItem,
-} from "@blueprintjs/core";
-import { IconNames } from "@blueprintjs/icons";
-import styled from "@emotion/styled";
-import DropboxChooser, {
-  Props as ChooserProps,
-} from "src/components/DropboxChooser";
-import DeleteDataset from "../../../DeleteDataset";
-import { Collection } from "src/common/entities";
-
-const DeleteButton = (props: MenuItemProps) => {
-  return (
-    <MenuItem
-      {...props}
-      shouldDismissPopover={false}
-      icon={IconNames.TRASH}
-      intent={Intent.DANGER}
-      text="Delete Dataset"
-    />
-  );
-};
-
-const UpdateButton = (props: Partial<MenuItemProps>) => {
-  return (
-    <MenuItem
-      {...props}
-      shouldDismissPopover={false}
-      icon={IconNames.EDIT}
-      intent={Intent.NONE}
-      text="Update Dataset"
-    />
-  );
-};
-
-interface Props {
-  collectionId: Collection["id"];
-  datasetId?: string;
-  isPublished: boolean;
-  revisionsEnabled: boolean;
-  onUploadFile: ChooserProps["onUploadFile"];
-  isLoading: boolean;
-}
-
-const StyledMenu = styled(RawMenu)`
-  border-radius: 3px;
-  padding: 8px;
-
-  & > li:last-child {
-    margin-bottom: 0;
-  }
-`;
+  Menu as StyledMenu,
+  MenuItem as StyledMenuItem,
+} from "src/views/Collection/components/CollectionActions/components/MoreDropdown/components/Menu/style";
+import {
+  DELETE_ICON_PROPS,
+  EDIT_ICON_PROPS,
+  MENU_PROPS,
+} from "src/views/Collection/components/CollectionActions/components/MoreDropdown/components/Menu/constants";
+import { MENU_ITEM_COLOR } from "src/views/Collection/components/CollectionActions/components/MoreDropdown/components/Menu/types";
+import { Icon } from "@czi-sds/components";
+import EditDataset from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset";
+import { isDeleteDatasetAvailable } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/components/Menu/utils";
+import { DROPDOWN_EDIT_DATASET } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/MoreDropdown/components/Menu/constants";
 
-const Menu = ({
+export default function Menu({
   collectionId,
-  datasetId = "",
-  isPublished,
-  revisionsEnabled,
-  onUploadFile,
-  isLoading,
-}: Props): JSX.Element => {
-  // A dataset may be deleted if the collection is private, or the dataset has not been previously
-  // published; where the published_at property is used to determine whether the dataset has been previously published.
-  const shouldShowDelete = !revisionsEnabled || !isPublished;
```

Comment:
`shouldShowDelete` logic is now encapsulated in a function [isDeleteDatasetAvailable](https://github.com/chanzuckerberg/single-cell-data-portal/pull/7234/files#diff-2b59f9159123e7170cd4b47f1e0bc5fb828848539c96d5c4d9292423fa59eb7dR27).